azerothcore/azerothcore-wotlk
# feat(Core/Packet): Implement AntiDOS protection from Trinity

## CHANGES PROPOSED:
-  https://github.com/TrinityCore/TrinityCore/commit/aa5bc37e9cf301b0917dc4fd319c91c13aff654d
-  https://github.com/TrinityCore/TrinityCore/commit/87e105c81836b46e2a307031bf0083fcc7f75f8b
-  https://github.com/TrinityCore/TrinityCore/commit/4aee86627e001314ec80655615676e9593a06462
-  https://github.com/TrinityCore/TrinityCore/commit/a4ba54fbdbb690377c873ea34f1d3541002872b5
-  https://github.com/TrinityCore/TrinityCore/commit/2f0f8f8018f4786b9e3e3eab9a9ec17c02668923
-  https://github.com/TrinityCore/TrinityCore/commit/c06dc7d37aa8c795ff505550a7bc388d2a66ecac
-  https://github.com/TrinityCore/TrinityCore/commit/b5b64a2cdd6ccc935610b090b42d695df083eebb
-  https://github.com/TrinityCore/TrinityCore/commit/f312b869417d5bc0ea10928284d040e04259c1d8
-  https://github.com/TrinityCore/TrinityCore/commit/1b47f3270ec0c02ececb6cf96da957aa96a739dc
-  https://github.com/TrinityCore/TrinityCore/commit/65917f52eaafcd157b68444f2ab084bb428e40b6
-  https://github.com/TrinityCore/TrinityCore/commit/8207aff0d6f3fe475d1f088a9c9f589792e008fa
-  https://github.com/TrinityCore/TrinityCore/commit/d52eb14bb6f0fbfb9d75cc0f16793eefb74666c7
-  https://github.com/TrinityCore/TrinityCore/commit/2ba05406ce0632d30498b05981a02f2b7d4f75da
-  https://github.com/TrinityCore/TrinityCore/commit/06962b16f147275cf3ef23caf2f41ce0d25c25ef
-  https://github.com/TrinityCore/TrinityCore/commit/15b1d4174017f59482bc5201367c01a9dc587733
-  https://github.com/TrinityCore/TrinityCore/commit/d8c3115bbd1b544d6eca5d9aae3dfbb547353703
-  https://github.com/TrinityCore/TrinityCore/commit/5bd7c295c0b0f2cb78345ed02dd37531a67348f5

Co-authored-by: jackpoz <giacomopoz@gmail.com>
Co-authored-by: Shauren <shauren.trinity@gmail.com>
Co-authored-by: Vincent-Michael <trinity.michael_vincent@gmx.eu>


## TESTS PERFORMED:
- Compiled under windows and played around the server a while.




## HOW TO TEST THE CHANGES:

Just play around in the server and see if everything is normal. To check the antidos system just flood packets to the server with some api.

## Target branch(es):








 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
